lib/utilities.js

```diff
@@ -74,15 +152,131 @@ class Utilities {
+			})
+			.finally(() => clearTimeout(requestTimeout));
+
+		let json;
+
+		try {
+			json = await response.json();
+		} catch (err) {
+			throw new Error(err);
+		}
+
+		if (Object.keys(json).length < 0)
+			throw new Error("ERROR: AniList API is down. Please refer to official channels for more information.");
+
+		const { errors } = json;
+		if (errors) return errors;
+
+		const { Media, Character, Staff, Page, Recommendation, Studio, User, Viewer, MediaListCollection } = json.data;
+
+		if (Media) return formatMedia(Media);
+
+		if (Character) {
+			Character.name = nameEditor(Character.name);
+			Character.media = Character.media.nodes;
+			return Character;
+		}
+
+		if (Staff) {
+			Staff.name = nameEditor(Staff.name);
+			Staff.staffMedia = edgeRemove(Staff.staffMedia.nodes);
+			Staff.characters = edgeRemove(Staff.characters.nodes);
+			return Staff;
+		}
+
+		if (Page) {
+			// For list of recent activities with getRecentActivity.
+			if (Page.activities) return Page.activities;
+
+			if (Recommendation) {
+				// For recommendation lists.
+				Recommendation.recommendations = Page.recommendations;
+				return Recommendation;
+			}
+
+			// For general searching
+			return Page;
+		}
+
+		if (Studio) {
+			Studio.media = edgeRemove(Studio.media.nodes);
+			return Studio;
+		}
+
+		if (User || Viewer) {
+			let userObj = User || Viewer;
+			if (userObj.statistics) {
+				// Move all names up a level.
+				userObj.statistics.anime.staff.forEach((e) => e.staff.name = e.staff.name.english);
+				userObj.statistics.anime.voiceActors.forEach((e) => e.voiceActor.name = e.voiceActor.name.english);
+				userObj.statistics.manga.staff.forEach((e) => e.staff.name = e.staff.name.english);
+			}
+
+			if (userObj.statistics && !userObj.avatar) return userObj.statistics;
+
+			// Move all node objects up one level.
+			userObj.favourites.anime = userObj.favourites.anime.nodes;
+			userObj.favourites.manga = userObj.favourites.manga.nodes;
+			userObj.favourites.characters = edgeRemove(userObj.favourites.characters.nodes);
+			userObj.favourites.staff = edgeRemove(userObj.favourites.staff.nodes);
+			userObj.favourites.studios = userObj.favourites.studios.nodes;
+
+			return userObj;
+		}
+
+		if (MediaListCollection) {
+			MediaListCollection.lists.forEach((list) =>
+				list.entries.map((entry) => {
+					// Media does not need to be formatted in a list query.
+					entry.dates = {
+						startedAt: convertFuzzyDate(entry.startedAt),
+						completedAt: convertFuzzyDate(entry.completedAt),
+						updatedAt: new Date(entry.updatedAt * 1000),
+						createdAt: entry.createdAt === 0 ? null : new Date(entry.createdAt * 1000)
+					};
+					["startedAt", "completedAt", "updatedAt", "createdAt"].forEach((e) => delete entry[e]);
+				})
+			);
+			return MediaListCollection.lists;
+		}
+
+		// If nothing matches, return collected data
+		return json.data;
+	}
+}
```

Comment:
AniList-Node uses semantic versioning so any major changes would make it a major update. Migrating the fetcher.js code into utilities.js is a major change in the internal structure so I would classify this as a v2.0.0 major update and not what was originally commented at the start which I would classify as a minor update.

Either revert this and the fetcher.js file or these changes will not be merged for a while due to this.